File: JavaScript/Stack/stackArray.js
class Stack {
    constructor() {
      // We can also use a linked list here
      this.items = [];
    }
  
    // Time complexity: O(1)
    push(item) {
      this.items.push(item);
    }
  
    // Time complexity: O(1)
    pop() {
      return this.items.pop();
    }
  
    // Time complexity: O(1)
    size() {
      return this.items.length;
    }
  
    // Time complexity: O(1)
    peek() {
      return this.items[this.items.length - 1];
    }
  }
  
  const stack = new Stack();
  stack.push(5);              // Time complexity: O(1)
  stack.push(12);             // Time complexity: O(1)
  stack.push(5);              // Time complexity: O(1)
  console.log(stack.peek());  // Time complexity: O(1)
  console.log(stack.pop());   // Time complexity: O(1)
  console.log(stack.peek());  // Time complexity: O(1)
  console.log(stack.size());  // Time complexity: O(1)
  

File: Python/Stack/stackArray.py
class Stack:
    def __init__(self):
        # We can also use a linked list here
        self.items = []

    # Time complexity: O(1)
    def push(self, item):
        self.items.append(item)

    # Time complexity: O(1)
    def pop(self):
        return self.items.pop()

    # Time complexity: O(1)
    def size(self):
        return len(self.items)

    # Time complexity: O(1)
    def peek(self):
        return self.items[-1]


stack = Stack()
stack.push(5)               # Time complexity: O(1)
stack.push(12)              # Time complexity: O(1)
stack.push(5)               # Time complexity: O(1)
print(stack.peek())         # Time complexity: O(1)
print(stack.pop())          # Time complexity: O(1)
print(stack.peek())         # Time complexity: O(1)
print(stack.size())         # Time complexity: O(1)
